src/utils/dialog/dialog-service.ts

```diff
@@ -0,0 +1,129 @@
+import Vue from 'vue';
+import { PluginObject } from 'vue/types/plugin';
+import { MDialog, MDialogState } from '../../components/dialog/dialog';
+
+
+declare module 'vue/types/vue' {
+    interface Vue {
+        $dialog: DialogService;
+    }
+}
+
+export class DialogService {
+
+    /**
+     *
+     * @param message the message of the confirmation dialog
+     * @param title a title label if any
+     * @param okLabel a cancel label if any
+     */
+    public alert(message: string, title?: string, okLabel?: string): Promise<boolean> {
+        let alertInstance: MDialog = new MDialog({
+            el: document.createElement('div')
+        });
+
+        document.body.appendChild(alertInstance.$el);
+        alertInstance.message = message;
+        alertInstance.okLabel = okLabel ? okLabel : undefined;
+        alertInstance.title = title ? title : '';
+        alertInstance.negativeLink = false;
+
+        return this.show(alertInstance).then((result) => {
+            alertInstance.$destroy();
+            return result;
+        });
+    }
+
+
+    /**
+     *
+     * @param message the message of the confirmation dialog
+     * @param title a title label if any
+     * @param okLabel a ok label if any
+     * @param cancelLabel a cancel label if any
+     */
+    public confirm(message: string, title?: string, okLabel?: string, cancelLabel?: string): Promise<boolean> {
+        let confirmInstance: MDialog = new MDialog({
+            el: document.createElement('div')
+        });
+
+        document.body.appendChild(confirmInstance.$el);
+        confirmInstance.message = message;
+        confirmInstance.type = MDialogState.Confirmation;
+        confirmInstance.title = title ? title : '';
+        confirmInstance.okLabel = okLabel ? okLabel : undefined;
+        confirmInstance.cancelLabel = cancelLabel ? cancelLabel : undefined;
+
+        return this.show(confirmInstance).then((result) => {
+            confirmInstance.$destroy();
+            return result;
+        });
+    }
+
+
+    /**
+     *
+     * @param mDialogInstance the MDialog instance
+     * @param rejectOnCancel if true, the promise will be rejected on cancel
+     */
+    public show(mDialogInstance: MDialog, rejectOnCancel?: boolean): Promise<boolean> {
+        return new Promise((resolve, reject) => {
+
+            let onOk: () => void = () => {
```

Comment:
Hmmm les nested functions ici je vois pas trop le gain?
1- Maintenant pour simuler du private on peut utiliser les classes avec le private
2- C'est pas optimal pour le heap javascript et la performance https://code.tutsplus.com/tutorials/stop-nesting-functions-but-not-all-of-them--net-22315
3- C'est pas des closures.

Ça aurait donc pu facilement être des fonctions privates sur ton DialogService.

Autre point, est-ce vraiment nécéssaire de resolve dans le nextick ici? Ça cause des problèmes d'update sinon?

Reply:
Merci pour les inputs, on fait le style guide bientôt pour modul, on pourra y ajouter ce point au niveau performance.

POur ce qui est de nexttick ca provient des fonctions alert.ts et confirm.ts fait par @simardo  , j'ai simplement refactorer en service pour éliminer la duplication et corrigé un problème un perf. Faudrait investiguer pourquoi c'était la.

Reply:
Ouais faute de le régler immédiatement est-ce qu'on pouvvait mettre un TODO : trouver pourquoi le nextick est nécéssaire?

Idéalement les nextick j'aime mieux qu'ils soient documentés
J'ai fait ça pcq XYZ
ou j'ai fait ça mais je vais y revenir plus tard pcq je pense que c'est pas normal d'en avoir besoin :)